client_test/stub_test.py

```diff
@@ -166,6 +169,22 @@ def test_serve(client):
     stub.serve(client=client, timeout=1)
 
 
+@pytest.mark.skipif(
+    sys.version.startswith("3.7"),
+    reason="AsyncMock only introduced in Python 3.8",
+)
+def test_serve_teardown(client, servicer):
+    from unittest.mock import AsyncMock, patch
+
+    stub = Stub()
+    fake_disconnect = AsyncMock()
+    with modal.client.Client(servicer.remote_addr, api_pb2.CLIENT_TYPE_CLIENT, ("foo-id", "foo-secret")) as client:
+        with patch.object(modal.app._App, "disconnect", fake_disconnect):
```

Comment:
Awesome with test!

Instead of mocking the app, you could probably just see if the `AppDisconnect` call was called [on the servicer fixture](https://github.com/modal-labs/modal-client/blob/main/client_test/conftest.py#L110)

Reply:
Cheers, I'll try that 👍

Reply:
A mock-y test that runs in CI is probably better than a non-mock test that doesn't run in CI, so I'll go back to that.

Reply:
Damn, the mock is flaky too.